### PR TITLE
[MBL-15716][Student] - Implement Homeroom E2E test

### PR DESCRIPTION
**Pull request by kdeakinstructure**
refs: MBL-15716
affects: Student
release note: none

Homeroom E2E test implementation.

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/student/src/androidTest/java/com/instructure/student/ui/e2e/HomeroomE2ETest.kt [hunk 1]
```diff
@@ -49,12 +56,78 @@ class HomeroomE2ETest : StudentTest() {
...
+        homeroomPage.assertAnnouncementDisplayed(
+            homeroomCourse.name,
+            homeroomAnnouncement.title!!,
+            homeroomAnnouncement.message!!
+        )
+
+        homeroomPage.assertCourseItemsCount(3) //gives back the number of courses under 'My Subject' list
+        homeroomPage.clickOnViewPreviousAnnouncements()
+        announcementListPage.assertToolbarTitle()
+        announcementListPage.assertAnnouncementTitleVisible(homeroomAnnouncement.title!!)
```
Comment (hermannakos) [inline apps/student/src/androidTest/java/com/instructure/student/ui/e2e/HomeroomE2ETest.kt:102]: The non-null assertion is unnecessary, there are a few others as well.

File: apps/student/src/androidTest/java/com/instructure/student/ui/e2e/HomeroomE2ETest.kt [hunk 2]
```diff
@@ -49,12 +56,78 @@ class HomeroomE2ETest : StudentTest() {
...
+        Espresso.pressBack()
+
+        var noHomeroomCourseTitleIndex = 1
+
+        for (i in 0 until 2) {
+                homeroomPage.assertCourseDisplayed(
+                    nonHomeroomCourses[i].name,
+                    homeroomPage.getStringFromResource(R.string.nothingDueToday),
+                    data.announcementsList[noHomeroomCourseTitleIndex++].title
```
Comment (hermannakos) [inline apps/student/src/androidTest/java/com/instructure/student/ui/e2e/HomeroomE2ETest.kt:111]: You can use i+1

File: apps/student/src/androidTest/java/com/instructure/student/ui/e2e/HomeroomE2ETest.kt [hunk 3]
```diff
@@ -49,12 +56,78 @@ class HomeroomE2ETest : StudentTest() {
...
+
+        homeroomPage.assertCourseItemsCount(3) //gives back the number of courses under 'My Subject' list
+        homeroomPage.clickOnViewPreviousAnnouncements()
+        announcementListPage.assertToolbarTitle()
+        announcementListPage.assertAnnouncementTitleVisible(homeroomAnnouncement.title!!)
+        Espresso.pressBack()
+
+        var noHomeroomCourseTitleIndex = 1
+
+        for (i in 0 until 2) {
```
Comment (hermannakos) [inline apps/student/src/androidTest/java/com/instructure/student/ui/e2e/HomeroomE2ETest.kt:107]: This should be `i in 0 until nonHomeroomCourses.size - 1`

**Review by hermannakos — CHANGES_REQUESTED** [review]

File: apps/student/src/androidTest/java/com/instructure/student/ui/e2e/HomeroomE2ETest.kt (hunk 3 above)
Comment (kdeakinstructure) [reply apps/student/src/androidTest/java/com/instructure/student/ui/e2e/HomeroomE2ETest.kt:107]: Nice catch, thx

**Comment by kdeakinstructure** [comment]
E2E test run: https://app.bitrise.io/build/5a99c8cd-bac3-445c-b699-6f2b957f7486#?tab=log

**Review by hermannakos — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**